pandas/io/parsers.py

```diff
@@ -420,7 +420,9 @@ def _validate_names(names):
     if names is not None:
         if len(names) != len(set(names)):
             raise ValueError("Duplicate names are not allowed.")
-        if not is_list_like(names, allow_sets=False):
+        if not is_list_like(names, allow_sets=False) and not isinstance(
```

Comment:
can you do
not (is_list_like or isintance....)

Reply:
@jreback Yup, done. Thanks for that!